benefits/oauth/views.py

```diff
@@ -47,8 +49,68 @@ def authorize(request):
     if token is None:
         logger.warning("Could not authorize OAuth access token")
         return redirect(ROUTE_START)
-    else:
-        # we are intentionally not storing anything about the user, including their token
-        logger.debug("OAuth access token authorized")
-        session.update(request, auth=True)
-        return redirect(ROUTE_CONFIRM)
+
+    logger.debug("OAuth access token authorized")
+
+    # we store the id_token in the user's session
+    # this is the minimal amount of information needed later to log the user out
+    session.update(request, oauth_token=token["id_token"])
+
+    return redirect(ROUTE_CONFIRM)
+
+
+def logout(request):
+    """View implementing OIDC and application sign out."""
+
+    # overwrite the oauth session token, the user is signed out of the app
+    token = session.oauth_token(request)
+    session.update(request, oauth_token=False)
+
+    route = reverse(ROUTE_POST_LOGOUT)
+    redirect_uri = _generate_redirect_uri(request, route)
+
+    logger.debug(f"OAuth end_session_endpoint with redirect_uri: {redirect_uri}")
+
+    # send the user through the end_session_endpoint, redirecting back to
+    # the post_logout route
+    return _deauthorize_redirect(token, redirect_uri)
+
+
+def post_logout(request):
+    """View routes the user to their origin after sign out."""
+
+    origin = session.origin(request)
+    return redirect(origin)
```

Comment:
> When the user makes it to the `/enrollment` phase, their session will indicate they have been verified as eligible. So on sign out, we can redirect them back into the `/enrollment` phase vs. making them restart with `/eligibility`.

Hmmmm. I know we had a lot of back and forth about what "logging out" means, but I would expect a "sign out" link to clear my logged-in state (session) on the current site (Benefits).

Also, the stored `oauth_token` from the auth server will no longer be valid, but that's still how we're tracking the auth state in Benefits. Therefore, once we get the logout link placed, that link will:

- Appear to users even though they logged out, and
- Not work if the token is rejected by the auth server
    - I haven't tested this, and even if it doesn't, I could see that changing in an upgrade.

Reply:
> I would expect a "sign out" link to clear my logged-in state (session) on the current site (Benefits).

Yes, the Benefits logged-in state is cleared: https://github.com/cal-itp/benefits/pull/436/files/2e291f2c50007e743fb19aba389bdee41149d4a8#diff-046f7cf43c214ae0ab405c72f331b032dd2c52f707e95f6fb9e33f35f07f7170R67

> Appear to users even though they logged out

I'm not sure where you're getting this from? #428 is pretty clear: *"Only display when auth is enabled, and the user is logged in"*

Reply:
> the Benefits logged-in state is cleared

D'OH. My bad, totally missed that. Sorry!